lib/pbench/server/api/resources/upload_api.py

```diff
@@ -60,12 +60,12 @@ def put(self, filename: str):
 
         if os.path.basename(filename) != filename:
             msg = "File must not contain a path"
-            self.logger.warning("{} for {}, user = {}", msg, filename, username)
+            self.logger.warning("{} for {!a}, user = {}", msg, filename, username)
             abort(HTTPStatus.BAD_REQUEST, message=msg)
 
         if not self.allowed_file(filename):
             self.logger.warning(
-                "Unsupported file extension for {}, user = {}", filename, username,
+                "Unsupported file extension for {!a}, user = {}", filename, username,
```

Comment:
It occurs to me this message is potentially a bit fragile. The method is named `allowed_file` and one could imagine, for example, expanding the function to check for legal characters in the string in addition to just checking the suffix, while the message (which is at best loosely coupled with the function) specifically says "file extension".

Not really a big deal at this point, but generalizing the message to better match the method name (or tightening the implied scope of the method name to match the message) might possibly reduce someone's confusion in the future.

Also, since you fall into a pattern later of user, controller, filename, it'd be more consistent to use user, filename here even though we haven't yet established controller.